apps/ejabberd/src/ejabberd_c2s.erl

```diff
@@ -2308,27 +2366,27 @@ check_privacy_and_route_or_ignore(StateData, From, To, Packet, Dir) ->
                                         Packet :: exml:element(),
                                         Dir :: in | out) -> any().
 check_privacy_and_route_or_ignore(Acc, StateData, From, To, Packet, Dir) ->
-    {Acc1, Res} = privacy_check_packet(Acc, To, Dir, StateData),
+    Acc1 = mongoose_acc:put(to_send, Packet, Acc),
+    {Acc2, Res} = privacy_check_packet(Acc1, To, Dir, StateData),
     case Res of
-        allow -> ejabberd_router:route(From, To, Packet);
-        _ -> ok
-    end,
-    Acc1.
+        allow -> ejabberd_router:route(From, To, Acc2);
+        _ -> Acc2
```

Comment:
You mean why don't we remove it? For me, it doesn't hurt to have it here, if only for introspection (see previous comment on handling cached valued and 'to_send').